internal/storage/db/internal/db.go

```diff
@@ -122,6 +128,33 @@ func (s *dbStorage) DeleteSchema(ctx context.Context, ids ...string) error {
 	return nil
 }
 
+func (s *dbStorage) LoadPolicy(ctx context.Context, policyKey string) (*policy.Wrapper, error) {
+	knv := strings.Split(policyKey, ".")
+	if len(knv) < policyKeyMinSubItem || len(knv) > policyKeyMaxSubItem {
+		return nil, fmt.Errorf("invalid format for policyKey")
+	}
+
+	version := ""
+	if len(knv) == policyKeyMaxSubItem {
+		version = knv[2]
+	}
```

Comment:
Why do we need to handle the case where `version` is not part of the policy key? Also, if the defaultversion is empty string, would it match anything?

Reply:
I forgot about derived roles. My bad.

Reply:
Yes, unfortunately derived roles doesn't have version field :( They caused some headaches with this implementation and also with the new implementation.